src/util/isPromise.ts

```diff
@@ -1,3 +1,3 @@
-export function isPromise(value: any): boolean {
-  return value && typeof value.subscribe !== 'function' && typeof value.then === 'function';
-}
\ No newline at end of file
+export function isPromise<T>(value: any | Promise<T>): value is Promise<T> {
```

Comment:
Would you able to share some details of
- what does generic <T> does do?
- reason to change return type from boolean?